protos/core.proto

```diff
@@ -424,4 +429,12 @@ message TableMetadata {
   Target target = 1;
   string type = 2; // "table" or "view"
   repeated Field fields = 3;
+  int64 lastUpdatedMillis = 4;
+}
+
+message PersistedTableMetadata {
+  TableMetadata table = 1;
+  string create_query = 2;
+  int64 lastUpdatedMillis = 3;
+  repeated PersistedTableMetadata dependencies = 4;
```

Comment:
Recursive protos are annoying, also this is slightly incorrect (one of these repeated objects shouldn't have any children itself).
I think you can avoid this here by breaking this into 2 messages

Reply:
broke into two message